src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -1729,7 +1729,8 @@ case class InternalGcsStorage(param: MergedParameters,
     val metadatas = generateFileTransferMetadatas(
       data, format, compress, directory, data.getNumPartitions)
 
```

Comment:
To Reviewer:
After enabling these tests, I find one issue for writing an empty DataFrame (partition count is 0) to snowflake on GCP.
Separately Jira is filed. The fix is one line fix, so, I just bundle the fix here because current test can test it.
  https://snowflakecomputing.atlassian.net/browse/SNOW-296768

Reply:
to write a empty data frame, we can create a dummy row first to make it not empty, and then filter this dummy row out, to make it empty again.

Reply:
It looks there are two cases for empty data frame.
Case 1: The partition count for the data frame is > 0, but the row counts for each partition is 0.
Case 2: The partition count is 0.

We have test cases for case 1. But we don't have test case for case 2. Below code can generate data frame without any partition. 
```
    val df = Seq.empty[(Int, String)].toDF("key", "value")

```